packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -175,20 +189,66 @@ export class Role extends Resource implements IRole {
           grantee,
           actions,
           resourceArns: [this.roleArn],
-          scope: this
+          scope: this,
         });
       }
+    }
 
-      /**
-       * Grant permissions to the given principal to pass this role.
-       */
-      public grantPassRole(identity: IPrincipal): Grant {
-        return this.grant(identity, 'iam:PassRole');
+    const roleAccount = parsedArn.account;
+
+    class MutableImport extends Import {
+      private readonly attachedPolicies = new AttachedPolicies();
+      private defaultPolicy?: Policy;
+
+      public addToPolicy(statement: PolicyStatement): boolean {
+        if (!this.defaultPolicy) {
+          this.defaultPolicy = new Policy(this, 'Policy');
+          this.attachInlinePolicy(this.defaultPolicy);
+        }
+        this.defaultPolicy.addStatements(statement);
+        return true;
+      }
+
+      public attachInlinePolicy(policy: Policy): void {
+        const policyStack = Stack.of(policy);
+        const policyAccount = policyStack.account;
+
+        // this is complicated, so I extracted the condition to a separate variable
+        // we should attach the policy if...
+        const shouldAttachPolicy =
+          // ...the policy is in the same stack as the role was imported into...
+          policyStack === scopeStack ||
+          // ...or the policy is from an env-agnostic stack...
+          Token.isUnresolved(policyAccount) ||
+          // ...or the ARN with which the role was imported with is dynamic...
+          Token.isUnresolved(roleAccount) ||
+          // ...or the policy is in the same account as the imported role
+          policyAccount === roleAccount;
```

Comment:
If we were to make the return values of `Fn` singletons (similar to what `CfnReference` does), we could short-circuit the "stack are the same objects" check and only do this check (since in that case, `Fn.awsAccount() would be === Fn.awsAccount()`).

Reply:
They actually are singletons - we don't have `Fn.awsAccount()` anymore, but `Aws.ACCOUNT_ID`. Changed.